azure/Guidelines.md

```diff
@@ -0,0 +1,864 @@
+
+> :white_check_mark: **DO** sort the collection in ascending order on an expression if "asc" or "desc" is not specified.
+> 
+> :white_check_mark: **DO** sort NULL values as "less than" non-NULL values.
+> 
+> :white_check_mark: **DO** sort items by the result values of the first expression, and then sort items with the same value for the first expression by the result value of the second expression, and so on.
+> 
+> :white_check_mark: **DO** use the inherent sort order for the type of the field. For example, date-time values should be sorted chronologically and not alphabetically.
+> 
+> :white_check_mark: **DO** respond with an error message as defined in the [Handling Errors](#Handling-errors) section if the client requests sorting by a field that is not supported by the operation.
+
+For example, to return all people sorted by name in ascending order:
+```http
+GET https://api.contoso.com/people?orderby=name
+```
+
+For example, to return all people sorted by name in descending order and a secondary sort order of hireDate in ascending order.
+```http
+GET https://api.contoso.com/people?orderby=name desc,hireDate
+```
+
+Sorting MUST compose with filtering such that:
+```http
+GET https://api.contoso.com/people?filter=name eq 'david'&orderby=hireDate
+```
+will return all people whose name is David sorted in ascending order by hireDate.
+
+##### Considerations for sorting with pagination
+
+> :white_check_mark: **DO** use the same filtering options and sort order for all pages of a paginated list operation response.
+
+##### skip
+> :white_check_mark: **DO** define the _skip_ parameter as an integer with a default and minimum value of 0.
+
+> :heavy_check_mark: **YOU MAY** allow clients to pass the _skip_ query parameter to specify an offset into collection of the first resource to be returned.
+##### top
+
+> :heavy_check_mark: **YOU MAY** allow clients to pass the _top_ query parameter to specify the maximum number of resources to return from the collection.
+
+If supporting _top_:
+> :white_check_mark: **DO** define the _top_ parameter as an integer with a minimum value of 1. If not specified, __top__ has a default value of infinity.
+> 
+> :white_check_mark: **DO** return the collection's _top_ number of resources (if available), starting from _skip_.
+
+##### maxpagesize
+
+> :heavy_check_mark: **YOU MAY** allow clients to pass the _maxpagesize_ query parameter to specify the maximum number of resources to include in a single page response.
+
+If supporting _maxpagesize_
+
+> :white_check_mark: **DO** define the _maxpagesize_ parameter as an optional integer with a default value appropriate for the collection.
+> 
+> :white_check_mark: **DO** make clear in documentation of the _maxpagesize_ parameter that the operation may choose to return fewer resources than the value specified.
+
+### API Versioning
+
+Azure services need to change over time. However, when changing a service, there are 2 requirements:
```

Comment:
recommend comparing this with RP documentation. Data plane need not be the same as control plane, but good to know where we're different.

Reply:
+1. We have taken an action item to do this.